pkgs/development/compilers/llvm/common/openmp/default.nix

```diff
@@ -25,114 +25,91 @@
 
 assert lib.assertMsg (ompdGdbSupport -> ompdSupport) "OMPD GDB support requires OMPD support!";
 
-stdenv.mkDerivation (
-  finalAttrs:
-  {
-    pname = "openmp";
-    inherit version;
+stdenv.mkDerivation (finalAttrs: {
+  pname = "openmp";
+  inherit version;
 
-    src =
-      if monorepoSrc != null then
-        runCommand "openmp-src-${version}" { inherit (monorepoSrc) passthru; } (
-          ''
-            mkdir -p "$out"
-          ''
-          + lib.optionalString (lib.versionAtLeast release_version "14") ''
-            cp -r ${monorepoSrc}/cmake "$out"
-          ''
-          + ''
-            cp -r ${monorepoSrc}/openmp "$out"
-          ''
-        )
-      else
-        src;
+  src =
+    if monorepoSrc != null then
+      runCommand "openmp-src-${version}" { inherit (monorepoSrc) passthru; } (''
+        mkdir -p "$out"
+        cp -r ${monorepoSrc}/cmake "$out"
+        cp -r ${monorepoSrc}/openmp "$out"
+      '')
+    else
+      src;
 
-    sourceRoot = "${finalAttrs.src.name}/openmp";
+  sourceRoot = "${finalAttrs.src.name}/openmp";
 
-    outputs = [ "out" ] ++ lib.optionals (lib.versionAtLeast release_version "14") [ "dev" ];
+  outputs = [
+    "out"
+    "dev"
+  ];
 
-    patchFlags = if lib.versionOlder release_version "14" then [ "-p2" ] else null;
+  patches =
+    lib.optional (lib.versionAtLeast release_version "15" && lib.versionOlder release_version "19") (
+      getVersionFile "openmp/fix-find-tool.patch"
+    )
+    ++ lib.optional (lib.versionOlder release_version "18") (
+      getVersionFile "openmp/gnu-install-dirs.patch"
+    )
+    ++ [
+      (getVersionFile "openmp/run-lit-directly.patch")
+    ];
 
-    patches =
-      lib.optional (lib.versionAtLeast release_version "15" && lib.versionOlder release_version "19") (
-        getVersionFile "openmp/fix-find-tool.patch"
-      )
-      ++ lib.optional (lib.versionAtLeast release_version "14" && lib.versionOlder release_version "18") (
-        getVersionFile "openmp/gnu-install-dirs.patch"
-      )
-      ++ lib.optional (lib.versionAtLeast release_version "14") (
-        getVersionFile "openmp/run-lit-directly.patch"
-      )
-      ++
-        lib.optional (lib.versionOlder release_version "14")
-          # Fix cross.
-          (
-            fetchpatch {
-              url = "https://github.com/llvm/llvm-project/commit/5e2358c781b85a18d1463fd924d2741d4ae5e42e.patch";
-              hash = "sha256-UxIlAifXnexF/MaraPW0Ut6q+sf3e7y1fMdEv1q103A=";
-            }
-          );
+  nativeBuildInputs = [
+    cmake
+    python3.pythonOnBuildForHost
+    perl
+  ]
+  ++ lib.optionals (lib.versionAtLeast release_version "15") [
+    ninja
+    pkg-config
+    lit
+  ];
 
-    nativeBuildInputs = [
-      cmake
-      python3.pythonOnBuildForHost
-      perl
-    ]
-    ++ lib.optionals (lib.versionAtLeast release_version "15") [
-      ninja
-    ]
-    ++ lib.optionals (lib.versionAtLeast release_version "14") [
```

Comment:
This can't be folded into the previous conditional in the LLVM 13 removal, but needs to be its own, unconditional list.

Reply:
Good catch, thank you. Will split this up between the two commits.